api/src/main/java/xtdb/query/Query.java

```diff
@@ -311,7 +311,12 @@ static Call call(String ruleName, List<Expr> args) {
         return new Call(ruleName, args, null);
     }
 
-    final class Join implements UnifyClause {
+    abstract class AJoin implements UnifyClause {
```

Comment:
If I don't implement `UnifyClause` at that level I need to write an extra deserializer for `join` and `left-join` which I tried to avoid in https://github.com/xtdb/xtdb/pull/3033/commits/bcd8375845d7ec19e9950bd4433e2d213dc44484. I can also just revert that.

Reply:
Addressed the interface bit.